### PR TITLE
cmake:replace custom_patch_target with PATCH_COMMAND

### PR DESCRIPTION
**Pull request by xuxin930**
- delete the .git condition that cause duplication of fetching stage.
- replace custom_patch_target with PATCH_COMMAND.
- remove custom_patch_target dependencies.

related to:https://github.com/apache/nuttx-apps/pull/1993
## Testing
fetch for the first time and compile repeated

### CODE REVIEW

File: libs/libxx/libcxx.cmake [hunk 1]
```diff
@@ -18,10 +18,28 @@
...
 # ##############################################################################
 
-if(NOT EXISTS ${CMAKE_CURRENT_LIST_DIR}/libcxx/.git)
+if(NOT EXISTS ${CMAKE_CURRENT_LIST_DIR}/libcxx)
 
   set(LIBCXX_VERSION 12.0.0)
 
+  set(LIBCXX_PATCH_COMMAND
+      patch -p0 -d ${CMAKE_CURRENT_LIST_DIR} <
+      ${CMAKE_CURRENT_LIST_DIR}/0001-Remove-the-locale-fallback-for-NuttX.patch
```
Comment (xuxin930) [inline libs/libxx/libcxx.cmake:27]: @anchao 
there is a code-style problem here.
if cmake-format is used, the statement will be modified to :
```cmake
set(LIBCXX_PATCH_COMMAND
    patch
    -p0
    -d
    ${CMAKE_CURRENT_LIST_DIR}
    <
    ....
)
``` 
readability gets worse.
should we also use whitelist for cmake files?

**Comment by xiaoxiang781216** [comment]
@xuxin930 please rebase your patch.

**Comment by xuxin930** [comment]
> @xuxin930 please rebase your patch.

DONE

File: libs/libxx/libcxx.cmake (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libxx/libcxx.cmake:27]: can we merge into one line?

Comment (anchao) [reply libs/libxx/libcxx.cmake:27]: You can locally disable and enable code formatting by using the special comments

```
# cmake-format: off
code block
# cmake-format: on
```

https://cmake-format.readthedocs.io/en/latest/format-features.html

Comment (xuxin930) [reply libs/libxx/libcxx.cmake:27]: > can we merge into one line?

i don't think it's good, if so, this line will be very long.

Comment (xuxin930) [reply libs/libxx/libcxx.cmake:27]: > https://cmake-format.readthedocs.io/en/latest/format-features.html

thanks a lot, this is very useful

**Review by acassis — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by raiden00pl — APPROVED** [review]
Great, thanks! cmake works as expected.

Tested with:
```
cmake -B build_cpunet1 -DBOARD_CONFIG=nrf5340-dk:rptun_cpunet -GNinja
cmake -B build_cpunet2 -DBOARD_CONFIG=nrf5340-dk:rptun_cpunet -GNinja
cmake -B build_cpunet3 -DBOARD_CONFIG=nrf5340-dk:rptun_cpunet -GNinja
cmake --build build_cpunet1 
cmake --build build_cpunet2
cmake --build build_cpunet3
```

**Pull request merged**